lambdas/service/app.py

```diff
@@ -222,44 +222,154 @@ def openapi_raw():
+    'progress': 'is a more concrete version of __queues__. Progress indicates the remainder of '
+                'transactions and documents to be indexed. Which gives the client insight into the progress of '
+                'indexing.\n',
+    **health_service_keys
 }
 
 
+def health_docs_spec(keys: dict) -> str:
+    return ''.join(f'* __{k}__ {v} ' for k, v in keys.items())
+
+
+def health_example_spec(keys: dict, up: Optional[bool] = None):
+    example = {key: {} for key in keys.keys()}
+    example['up'] = up
+    return example
+
+
+def health_spec(health_keys):
+    return {
+        'responses': {
+            f'{200 if up else 503}': {
+                'description': f'Service component is {"" if up else "un"}reachable.'
+                               f'\n\nResponse keys:\n\n{health_docs_spec(health_keys)}' if up else '',
+                'content': {
+                    'application/json': {
+                        'example': {
+                            **health_example_spec(health_keys, up)
+                        },
+                        'schema': schema.object(
+                            additional_properties=schema.object(
+                                additional_properties=True,
+                                up=schema.enum(up)
+                            ),
+                            up=schema.enum(up)
+                        )
+                    }
+                }
+            } for up in [True, False]
+        },
+        'tags': ['Auxiliary']
+    }
+
+
+def health_controller():
+    return HealthController(lambda_name='service')
+
+
 @app.route('/health', methods=['GET'], cors=True, method_spec={
-    **health_spec,
-    'summary': 'List health information for webservice'
+    'summary': 'Complete health check',
+    'description': format_description('''
+        Health check of the service and all resources it depends on.
+        This may take long time to complete and exerts considerable load on the service.
+        For that reason it should not be requested frequently or by automated monitoring
+        facilities that would be better served by the
+        [`/health/fast`](#operations-Auxiliary-get_health_fast) endpoint.
+    '''),
+    **health_spec(health_all_keys)
 })
+def health():
+    return health_controller().full_response()
+
+
+@app.route('/health/basic', methods=['GET'], cors=True, method_spec={
+    'summary': 'Basic health check',
+    'description': format_description(f'''
+        Health check of only the REST API itself, excluding other resources
+        the service depends on. A 200 response indicates that the service is
+        reachable via HTTP(S) but nothing more.
+    '''),
+    **health_spec(health_up_key)
+})
+def basic_health():
+    return health_controller().basic_response()
+
+
+@app.route('/health/cached', methods=['GET'], cors=True, method_spec={
+    'summary': 'Cached health check for continuous monitoring',
+    'description': format_description('''
```

Comment:
Wrap all your descriptions at column 80. Use a tool for that. PyCharm can wrap text blocks for you.